libsolidity/analysis/TypeChecker.cpp

```diff
@@ -547,8 +547,8 @@ bool TypeChecker::visit(FunctionDefinition const& _function)
 
 		if (_function.isConstructor())
 			m_errorReporter.typeError(6482_error, _function.location(), "Constructor cannot be defined in interfaces.");
-		else if (_function.visibility() != Visibility::External)
-			m_errorReporter.typeError(1560_error, _function.location(), "Functions in interfaces must be declared external.");
+		else if (_function.visibility() != Visibility::External || _function.visibility() != Visibility::Public)
```

Comment:
> I'm not sure why, but some test fail ... can you help me identify why ?

No visibility will match here. This should be:

```suggestion
		else if (_function.visibility() != Visibility::External && _function.visibility() != Visibility::Public)
```